### DISCUSSION TITLE
Nightscout stuck on loading.

### DISCUSSION THREAD

**Discussion by BriarHoper** [NightscoutFoundation/xDrip#3484] [Q&A]
We recently changed from AAPS to IAPS. The data uploaded through iAPS to Nightscout fine for a few days. Then it started being very slow and only intermittently having CGM numbers. Now when I try to go the Nightscout site it is always stuck on the Loading the Client page with the Nightscout logo. When it was slow I tried using Admin tools to delete data, but it never actually seemed to work, at least the data size never decreased. It was around 83% last time I could see anything a week or so ago. Tried the troubleshooting pages. FreeDNS is current, Google cloud says active. What should I try next?

**Comment by Navid200** [comment]
83%?  That cannot be!

Can you please post a screenshot of the status page?
https://navid200.github.io/xDrip/docs/Nightscout/Status.html

**Reply by BriarHoper** [reply]
![Nightscout loading](https://github.com/NightscoutFoundation/xDrip/assets/68034261/67aa5592-8f8e-4c0c-937c-52bb499dc4a6)
Did you mean this? or the Google cloud page? I can no longer ever get past this loading page on Nightscout.

**Comment by BriarHoper** [comment]
I think this is what you want. Took a while to load.
Briar

**Reply by BriarHoper** [reply]
![NIghtscout status](https://github.com/NightscoutFoundation/xDrip/assets/68034261/0fb14f7a-603c-47fc-8a32-281239d3fa15)

**Reply by Navid200** [reply]
Please restart the server:  https://navid200.github.io/xDrip/docs/Nightscout/Restart.html

Then, please post a screenshot of the status page again.

**Reply by Navid200** [reply]
There have been important updates we have made since the platform you are running.  
Let's sort out the current problem.
Then, I suggest you switch to our stable branch.  I am happy to walk you through it.

**Reply by BriarHoper** [reply]
OK, I did what you suggest and Nightscout came back up.  Do you still want the screenshot? I would be thrilled to have help updating.

**Reply by Navid200** [reply]
What does it report as the database size now?

**Reply by BriarHoper** [reply]
![status nightscout](https://github.com/NightscoutFoundation/xDrip/assets/68034261/0e70bb81-07d7-4a15-b272-4f5bddd9a64c)
Is it the 84% or am I looking at the wrong thing?

**Reply by Navid200** [reply]
No, you are right.  But, please do me a favor.
Please check your variables as explained below:
https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables.html
You can just use the text editor just to see the variables.

Then, take a screenshot of the variables.
Edit the screenshot and blackout the variable titled API_SECRET.
Other than that, there should be no other passwords.  But, if you have any other passwords in your variables, please blackout those as well.
Now, please post the edited screenshot so that I can see your variables minus any private variables.

**Reply by Navid200** [reply]
You don't have to do all of that.
Can you look at the variables page here?  https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables.html

Do you see the heading defaults?
Do you see DBSIZE_MAX?
What is yours set to?

**Reply by BriarHoper** [reply]
I don't see those in the variables. Here is the webbrowser version.
![Uploading variables3.jpg…]()

**Comment by BriarHoper** [comment]
![variables2](https://github.com/NightscoutFoundation/xDrip/assets/68034261/5a5fc65b-7bb4-4b5d-bd7b-72e12318d90a)

**Reply by Navid200** [reply]
I think that explains it.
Please add the following to your variables.

export DBSIZE_MAX=”15000”

After saving, please restart the server again: https://navid200.github.io/xDrip/docs/Nightscout/Restart.html

Now, what does your Nightscout show for disk space?

**Reply by BriarHoper** [reply]
I haven't figured out how to add this to the variables from the directions. How do I add this?

**Reply by Navid200** [reply]
Something worries me here.
Who created this setup for you?
I am worried that someone has created this with the intent or even successfully based on an external database.

So, it is best not to make any changes until we figure out exactly what is going on here.

If you had created this setup using our guide, you would have had that variable already.

**Reply by BriarHoper** [reply]
I did create this myself using the guide. Here is the current status.
![new status](https://github.com/NightscoutFoundation/xDrip/assets/68034261/a119eddd-1166-414d-a142-427908b10019)

**Reply by BriarHoper** [reply]
I figured out just to type in the change. I am not savvy enough to make my own changes to the programming so I'm not sure where things got changed in setup.

**Reply by Navid200** [reply]
Someone must have changed the variables.  
If you add the one I suggested and restart the server, Nightscout should show the correct database size.

**Reply by BriarHoper** [reply]
Should I be worried about this? I have never entered the variables page that I am aware of. It's freaking me out a bit now.

**Reply by Navid200** [reply]
Either you edited it or someone else who you asked to help you and gave permission to.
Is it possible that you have forgotten?  It happens to all of us.
Or, someone may have suggested to you to overwrite all the variables with a new set.

**Reply by BriarHoper** [reply]
I don't remember getting help, but It's been a while so maybe I did. I restarted the machine. The Nightscout page still says 84% although the status says 40%.

**Reply by Navid200** [reply]
If you created this machine more than 2 years ago, that's another possibility.  Those variables were added after that time.

Are you sure you added the variable?
Are you sure you restarted the server after?

**Reply by BriarHoper** [reply]
I typed the variable into the terminal thing that opened when I opened the text variables option. I didn't know if I had to hit save or just type it in, so if I had to do something else, I didn't do it. I definitely rebooted after, the site was down for a few minutes while it did.

**Reply by Navid200** [reply]
If you created the machine before December 2, 2022, it makes sense because we added that variable after that date.

If you look "Using a text editor" on the following page, I have explained how to edit and save variables.

https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables.html

**Reply by BriarHoper** [reply]
OK, that did it. Down to 2%. I did make this early on in the Google Cloud option when we were going to lose whatever the option before that was, so that's probably it. Thank you so much.

**Reply by Navid200** [reply]
You're welcome.

If you like, you can switch to our stable release.
It takes a few steps.  We can do it now.  Or, if you prefer we can do it another time if you are tired now.

**Reply by BriarHoper** [reply]
I would like to do that, but if you're busy that's ok. I appreciate the time you've already spent on this.

**Reply by Navid200** [reply]
No, I'm not busy.

Go to the Google Cloud Setup submenu.  
Do you see an option "Bootstrap the stable release"?

If yes, run it.  It will ask you to confirm that you want to run it even though you have an existing setup.  Approve.
Tell me when it is done.

If Bootstrap the stable release option does not exist, let me know.

**Reply by BriarHoper** [reply]
![status after bootstrap](https://github.com/NightscoutFoundation/xDrip/assets/68034261/7f1ac162-4897-4951-b214-bacc4ffa9930)

**Reply by Navid200** [reply]
Did you also run phase 2?

**Reply by Navid200** [reply]
I was going to explain how to complete phase 2!
Well done!

**Reply by BriarHoper** [reply]
No, you said phase one and show you status so that is all I have done. Should I run phase 2?

**Reply by Navid200** [reply]
Well, I am surprised.
Did you reboot the server?

**Reply by BriarHoper** [reply]
Yes, I rebooted the server.

**Reply by BriarHoper** [reply]
Now I'm second guessing myself. I'm pretty sure I rebooted the server but I'm not sure.

**Reply by Navid200** [reply]
Is Nightscout running?

**Reply by BriarHoper** [reply]
Yes

**Reply by Navid200** [reply]
Please restart the server again to be sure.

**Reply by BriarHoper** [reply]
![new status again](https://github.com/NightscoutFoundation/xDrip/assets/68034261/17a381a4-cffb-42fc-ad28-bb79c027bd2a)

**Reply by Navid200** [reply]
If you choose option 2 on the status page, it will take you to a page showing you 4 items.
Do you see the all 4 items populated, Hostname, API_SECRET, FreeDNS user ID, and FreeDNS password?

Or, are only 2 of them populated?

**Reply by BriarHoper** [reply]
All 4 are populated. However, I see a mooo.com address that I thought was the one previous to the hopto.com address. I used to access the Nightscout by going to the mooo.com that was the free one you had to update every month. I have continued to keep the address active for some unknown reason. I followed the directions to change over when it came out. Now I access it at a hopto.com address. Not sure if there is some problem here.

**Reply by Navid200** [reply]
I'm not sure what problem you are referring to.
There is nothing wrong with mooo.com.

If the site shown on that page is mooo.com, that is your hostname.
If you want to change it, you can.

The one we had to update once a month was noip.com.

**Reply by BriarHoper** [reply]
Oh, I just checked. My No-IP.com is the hopto.com. It is where I have been accessing my nightscout. It is still working. I just checked the mooo.com and it works too. Is this a problem?

**Reply by Navid200** [reply]
Possibly!
We switched to FreeDNS in order to avoid having to use noip.com.
If you are using both, it is just looking for trouble.  No one should use noip.com any longer.
Is there a reason you don't want to switch to FreeDNS?

**Comment by BriarHoper** [comment]
I did that and it seems to have worked, I'm back to the status page.

**Reply by Navid200** [reply]
Now, from the same Google Cloud Setup submenu, run installation phase 1.
Let it run and complete and then, restart the server.
Then, please show me the status page again.

**Comment by BriarHoper** [comment]
I followed all the directions to switch to Freedns. That is what is on the
status #2. I think I had just accidentally used the old hopto address and
it still worked so I assumed it was the current one. Is there a way to get
it not to work? Or is it just redirecting to a different address for the
nightscout site?

**Reply by Navid200** [reply]
I am not sure what you have done and how you have kept your noip.com not expect a login once a month.
Whatever you are using to keep login in, just stop it.

And use the FreeDNS hostname.
Your FreeDNS hostname is shown on the status page.  Or, you can just log in to your FreeDNS account and go to the subdomain page to see your subdomains.  One of them is the one you are using.
You should s top using noip.com.

We never gave instructions to keep using noip.com.
I don't know anything about hopto or anything like that.

**Reply by Navid200** [reply]
The link you have included in your post goes to an old pull request.
It is getting very confusing now.
Do you need a break?

**Reply by Navid200** [reply]
Can I ask you not to use email to post here?  That may be part of the problem.  Your responses keep jumping from one branch to another.
Would you please log into GitHub and post in the discussions?

**Reply by BriarHoper** [reply]
I am doing fine. I will stop using the no-ip. It was not my intention to keep using it. I have my Free DNS site and it works fine.  As far as the update goes. Do I still need to do phase 2? The site is working so I can come back to this later if that is better for you.

**Reply by BriarHoper** [reply]
I am in Github now.

**Reply by Navid200** [reply]
You can continue to use your hopto setup.
If you choose to do that, it may be a good idea, to take a good set of notes now and save it in a place you can find easily the next time you may run into trouble so that you can easily see what you are doing.

If you want to follow our standard procedure, you should change your uploader to upload to mooo.com.
And you should change all your followers to use mooo.com.
And you should stop using hopto and noip.com.  You should stop your uploader from uploading to upto and you should stop your followers from reading from hopto.

I am not asking you to do one or the other.  It is up to you.

No, I don't believe you need to do phase 2.  Everything seems to be working fine now.

**Reply by BriarHoper** [reply]
Thank you so much. I have already changed everything to the mooo.com and will discontinue the other. I just thought it wouldn't work anymore so that must be the new one. I really appreciate your time today!. Thank you again. I have a much better understanding of (at least some parts) of the process now.

**Reply by Navid200** [reply]
You're welcome.
Please let me know if anything doesn't work later.

**Reply by lostboy86** [reply]
Sorry, i have the same problem, but i have follow the setup for new update for Ubuntu 24.
With the previous setup no problem, i also delete the old vm, all ns work normally with aaps, or if i point the url to the api v1 json (treatments, profiles, etc) but the gui of NS stuck on loading page.

I also try everything : 

- switch to the dev branch
- switch to custom branch
- switch to normal branch
- boostrap, phase 1, phase 2 for every branch
- update the platform

No solution.
I have all four parameters populated (api secret, freedns hostname, etc)
Is not the first time to follow your guide, i setup many nightscout follow your fantastic guide in this two year, but now i'm stuck with this Loading page.

Can you help me to point a solution?
Thanks a lot

**Reply by Navid200** [reply]
@lostboy86 You have posted in a 1-year-old thread that had already been addressed.  
Why didn't you open a new thread?

**Reply by lostboy86** [reply]
Sorry, i see this is not closed, and i was thinking not duplicated an issue.
Double sorry

**Reply by Navid200** [reply]
These are discussions, not issues.  
But, still when you post in a discussion, everyone who has ever posted in it will get a notification by default.
And we don't close discussions.
There is nothing wrong with posting here.  But, it will be easier for everyone else who may be experiencing a similar problem to follow if you open a new one.
Thanks

**Reply by Navid200** [reply]
@lostboy86 There is a new Ubuntu 24 image on Google Cloud.
It is possible that there is an incompatibility.
I am checking it out.  It may take a while until I have a result.

**Comment by lostboy86** [comment]
Thanks a lot, yes i update with latest Ubuntu 24, this issue occur for this
new setup.

Previous, with guide for Ubuntu 20, no problem

Il lun 14 apr 2025, 18:17 Navid ***@***.***> ha scritto:

**Reply by Navid200** [reply]
My new install eventually came up.  I just needed to wait a while.
If you are still having problems, please start a new thread and include a screenshot of the status page.
Please don't post in this old thread.
